### PR TITLE
feat: DEV-2424: Key commands taxonomy

### PR DESCRIPTION
**Pull request by Travis1282**
Add key commands to taxonomy drop down to allow for mouseless interactions

### CODE REVIEW

File: src/components/Taxonomy/Taxonomy.tsx [hunk 1]
```diff
@@ -454,14 +451,47 @@ const Taxonomy = ({
...
+        if (index >= 0) shiftFocus(index, 1);
+        if (searchInput === focusedElement) shiftFocus(0, 0);
+        break;
+      case "ArrowUp":
+        if (index > 0) shiftFocus(index, -1);
+        else if (index === 0) searchInput && searchInput.focus();
+        break;
+      case "Shift":
+        if (index >= 0) focusedElement.parentNode?.parentNode?.toggle(focusedElement.id);
+        if (searchInput) searchInput.focus();
```
Comment (hlomzik) [inline src/components/Taxonomy/Taxonomy.tsx:489]: why do we focus right after opening the subtree?

File: src/components/Taxonomy/Taxonomy.tsx [hunk 2]
```diff
@@ -454,14 +451,47 @@ const Taxonomy = ({
...
+          searchInput && searchInput.focus();
+        }
+        if (index >= 0) shiftFocus(index, 1);
+        if (searchInput === focusedElement) shiftFocus(0, 0);
+        break;
+      case "ArrowUp":
+        if (index > 0) shiftFocus(index, -1);
+        else if (index === 0) searchInput && searchInput.focus();
+        break;
+      case "Shift":
```
Comment (hlomzik) [inline src/components/Taxonomy/Taxonomy.tsx:487]: Left/Right to close/open subtrees

File: src/components/Taxonomy/Taxonomy.tsx [hunk 3]
```diff
@@ -454,14 +451,47 @@ const Taxonomy = ({
...
-      document.body.addEventListener("keydown", onEsc);
-    } else {
+    document.body.addEventListener("click", onClickOutside, true);
+    document.body.addEventListener("keydown", onKeyDown);
+    return () => {
       document.body.removeEventListener("click", onClickOutside);
-      document.body.removeEventListener("keydown", onEsc);
-    }
-  }, [isOpen]);
+      document.body.removeEventListener("keydown", onKeyDown);
```
Comment (hlomzik) [inline src/components/Taxonomy/Taxonomy.tsx:458]: it's better to move declaration before use

File: src/components/Taxonomy/Taxonomy.tsx [hunk 4]
```diff
@@ -188,75 +188,77 @@ const Item: React.FC<RowProps> = ({ style, item, dimensionCallback, maxWidth }:
...
   return (
-    <div ref={itemContainer} style={{ paddingLeft: padding, maxWidth, ...style, width: 'fit-content' }}>
+    <div ref={itemContainer} style={{ paddingLeft: padding, maxWidth, ...style, width: "fit-content" }}>
       {!isAddingItem ? (
-        <div className={[styles.taxonomy__item, customClassname].join(" ")} >
-          <div className={styles.taxonomy__grouping} onClick={() => toggle(id)}>
-            <LsChevron stroke="#09f" style={arrowStyle} />
+        <>
+          <div className={styles.taxonomy__measure}>
+            <label>{name}</label>
```
Comment (hlomzik) [inline src/components/Taxonomy/Taxonomy.tsx:204]: don't forget to merge `master` to have `htmlFor` in `label`

File: src/components/Taxonomy/Taxonomy.tsx [hunk 5]
```diff
@@ -454,14 +451,47 @@ const Taxonomy = ({
...
+        break;
+      case "ArrowUp":
+        if (index > 0) shiftFocus(index, -1);
+        else if (index === 0) searchInput && searchInput.focus();
+        break;
+      case "Shift":
+        if (index >= 0) focusedElement.parentNode?.parentNode?.toggle(focusedElement.id);
+        if (searchInput) searchInput.focus();
+        break;
+      default:
```
Comment (hlomzik) [inline src/components/Taxonomy/Taxonomy.tsx:491]: also it would be cool to focus `searhInput` whenever user presses some alpha-digital key

**Review by hlomzik — CHANGES_REQUESTED** [review]

File: src/components/Taxonomy/Taxonomy.tsx (hunk 1 above)
Comment (Travis1282) [reply src/components/Taxonomy/Taxonomy.tsx]: the focus is lost when the sub tree is re rendered. I'm open to suggestions

File: src/components/Taxonomy/Taxonomy.tsx (hunk 2 above)
Comment (Travis1282) [reply src/components/Taxonomy/Taxonomy.tsx]: I have changed this to a right click

File: src/components/Taxonomy/Taxonomy.tsx (hunk 3 above)
Comment (Travis1282) [reply src/components/Taxonomy/Taxonomy.tsx]: not sure I follow

File: src/components/Taxonomy/Taxonomy.tsx (hunk 5 above)
Comment (Travis1282) [reply src/components/Taxonomy/Taxonomy.tsx:478]: sure, which one?

**Review by hlomzik — APPROVED** [review]

**Pull request merged**